bundler/lib/bundler/ui/shell.rb

```diff
@@ -130,7 +141,7 @@ def tell_me(msg, color = nil, newline = nil)
       def tell_err(message, color = nil, newline = nil)
         return if @shell.send(:stderr).closed?
 
-        newline ||= !message.to_s.match?(/( |\t)\Z/)
+        newline = !message.to_s.match?(/( |\t)\Z/) if newline.nil?
```

Comment:
There was a bug in the `tell_err` method where passing `false` for `newline` was being ignored. I fixed this and added a corresponding spec.

Reply:
Nice catch, thank you!

Normally I'd ask moving the bug fixes to separate PRs so that they show up in the changelog separately (our release scripts scan the merged PRs to build the changelog from PR titles). But since I don't think anybody is using these modes, I think it's fine to keep them "hidden" in this PR.

But maybe split them to separate commits, so the change actually implementing the change to `bundle lock --print` is more clear when you travel git history?

Reply:
OK, I've split the small fixes into separate commits. ✔️